src/utils/resources.py

```diff
@@ -146,15 +147,22 @@ class AppResource:
     def __init__(self, properties: Dict[str, Any], app: str, manager=None):
         self.app = app
         self.manager = manager
+        properties = self.default_properties | properties
 
-        for key, value in self.default_properties.items():
-            if isinstance(value, str):
-                value = value.replace("__APP__", self.app)
-            setattr(self, key, value)
+        replacements: dict[str, str] = {
+            "__APP__": self.app,
+            "__YNH_ARCH__": system_arch(),
+            "__YNH_DEBIAN_VERSION__": debian_version(),
+        }
```

Comment:
Wokay but this just feels super convoluted to achieve this ... Can't we stick to the current code and add a `.replace()` for `__YNH_ARCH__` and another for `__YNH_DEBIAN_VERSION__` ? x_x

Reply:
Yes indeed we could.

The idea of my first comment was to have a dict that would be populated resource after resource.